### PR TITLE
fix(AsusWMI): Use both WMI methods

### PR DESCRIPTION
**Pull request by pastaq**
- Revert commit be7c492.
- Remove dangerous unwraps from firmware_attributes usage.
- Use FirmwareAttributes as an Option, add RogPlatform as fallback.

### CODE REVIEW

File: src/performance/gpu/asus/asus_wmi.rs [hunk 1]
```diff
@@ -31,7 +40,24 @@ impl AsusWmi {
...
+                        AttrValue::Integer(value) => Ok(value as f64),
+                        _ => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
+                    },
+                    Err(_) => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
+                },
+                None => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
+            }
+        } else {
+            Ok(self.platform.get_ppt_pl1_spl()? as f64)
+        }
```
Comment (ShadowApex) [inline src/performance/gpu/asus/asus_wmi.rs:60]: You can use `let Some(firmware_attributes) = ... else {}` pattern to reduce the amount of nesting here:

```suggestion
        let Some(firmware_attributes) = &self.attributes else {
            return Ok(self.platform.get_ppt_pl1_spl()? as f64);
        };
        match firmware_attributes
            .attributes()
            .iter()
            .find(|a| a.name() == "ppt_pl1_spl")
        {
            Some(attribute) => match attribute.current_value() {
                Ok(attr_value) => match attr_value {
                    AttrValue::Integer(value) => Ok(value as f64),
                    _ => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
                },
                Err(_) => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
            },
            None => Err(TDPError::FailedOperation("Failed to read SPL.".to_string())),
        }
```

File: src/performance/gpu/asus/asus_wmi.rs [hunk 2]
```diff
@@ -47,9 +73,31 @@ impl AsusWmi {
...
+                },
+
+                None => return Err(TDPError::FailedOperation("Failed to set SPL.".to_string())),
+            }
+        } else {
+            self.platform.set_ppt_pl1_spl(value as u8)?;
+        }
 
-        // Set the boost back to the expected value with the new TDP
+        // Set the boost back to the expeted value with the new TDP
```
Comment (ShadowApex) [inline src/performance/gpu/asus/asus_wmi.rs:100]: lol

File: src/performance/gpu/asus/asus_wmi.rs [hunk 3]
```diff
@@ -47,9 +73,31 @@ impl AsusWmi {
...
+                        return Err(TDPError::FailedOperation(format!(
+                            "Failed to set SPL: {e:}"
+                        )));
+                    }
+                },
+
+                None => return Err(TDPError::FailedOperation("Failed to set SPL.".to_string())),
+            }
+        } else {
+            self.platform.set_ppt_pl1_spl(value as u8)?;
```
Comment (ShadowApex) [inline src/performance/gpu/asus/asus_wmi.rs:97]: I don't see how this case will ever be hit. If the "asus-wmi" module is found, the `attributes` property will always be `Some`.

File: src/performance/gpu/asus/asus_wmi.rs [hunk 4]
```diff
@@ -78,15 +156,33 @@ impl AsusWmi {
...
             }
         };
-        if (stapm + value) < 1.0 || (stapm + value) > u8::MAX as f64 {
-            return Err(TDPError::InvalidArgument(
-                "Combined TDP + Boost value must be between 1 and 255".to_string(),
-            ));
-        }
-        let boost = (stapm + value) as u8;
+
+        let sppt_val = (stapm + value) as i32;
```
Comment (ShadowApex) [inline src/performance/gpu/asus/asus_wmi.rs:160]: Shouldn't we do some negative value validation here?

File: src/performance/gpu/asus/asus_wmi.rs (hunk 3 above)
Comment (pastaq) [reply src/performance/gpu/asus/asus_wmi.rs:97]: asus-wmi doesn't provide attributes, asus-armoury does. It depends on asus-wmi.

Comment (ShadowApex) [reply src/performance/gpu/asus/asus_wmi.rs:97]: Look at how the `AsusWmi` object is created in the constructor. If `RogPlatform::new()` successfully returns, then `self.attributes` is set to `Some`. So this case will never get hit.

**Comment by pastaq** [comment]
Seems to work. I can't set boost to 0 but I think that's an issue with OGUI.

OGUI also spams everything 3x when setting things, we should look into that.

**Review by ShadowApex — APPROVED** [review]

**Pull request merged**